pyproject.toml

```diff
@@ -6,7 +6,7 @@ requires = [
     "packaging",
     "setuptools>=61",
     "setuptools-scm>=8.0",
-    "torch == 2.4.0",
+    "torch == 2.5.0 --extra-index-url https://download.pytorch.org/whl/cu121",
```

Comment:
Given @youkaichao's comment:
>the cuda version should not matter that much. I think our current pipeline should still work even if pytorch itself is built against cuda 12.4 .

We should consider ditching the `--extra-index-url`. Perhaps this should be configurable, but one thing to note is that 2:4 sparse fp8 will require the Pytorch version that's built with 12.4